include/deal.II/grid/grid_tools.h

```diff
@@ -157,10 +158,27 @@ namespace GridTools
    * get the same result using <code>cell-@>measure()</code>, but this
    * function also works for cells that do not exist except that you make it
    * up by naming its vertices from the list.
+   *
+   * Instantiations of this function are available for the <tt>dim !=
+   * spacedim</tt> cases, but these just raise exceptions. They are only
+   * provided as an aid for generic programming.
+   */
+  template <int dim, int spacedim>
```

Comment:
I tried that and it works with the library, but does not work with code that uses `cell_measure` with no template arguments for C++03 code. The function could previously do that so I would prefer to keep it (in the albeit unlikely case that a C++03 user is using this somewhat obscure function).

I want to try a few more things before admitting that we have to add a slightly breaking change.